exonum-java-binding/core/rust/src/storage/proof_map_index.rs

```diff
@@ -406,13 +393,34 @@ pub extern "system" fn Java_com_exonum_binding_core_storage_indices_ProofMapInde
     key: jbyteArray,
 ) -> Handle {
     let res = panic::catch_unwind(|| {
-        let key = convert_to_key(&env, key)?;
-        let iter = match *handle::cast_handle::<IndexType>(map_handle) {
-            IndexType::SnapshotIndex(ref map) => map.iter_from(&key),
-            IndexType::ForkIndex(ref map) => map.iter_from(&key),
+        let key = env.convert_byte_array(key)?;
+        let iter_handle = match *handle::cast_handle::<IndexType>(map_handle) {
+            IndexType::SnapshotIndex(ref index) => match index {
+                Index::Raw(map) => handle::to_handle(Iter::Raw(PairIter::new(
+                    &env,
+                    map.iter_from(&key.to_raw()),
+                    JAVA_ENTRY_FQN,
+                )?)),
+                Index::Hashed(map) => handle::to_handle(Iter::Hashed(PairIter::new(
+                    &env,
+                    map.iter_from(&key),
+                    JAVA_ENTRY_FQN,
```

Comment:
Looks like last argument of PairIter::new is always the same now, so it can be removed

Reply:
`PairIter` is used not only for `ProofMapIndexProxy`.

Reply:
I see, the usage in value_set_index has a different argument. All other are the same though, so we can still think about helper function